### PR TITLE
RIUI-417 Add dates to dashboard in correct format

### PR DESCRIPTION
**Pull request by Louisblack**
Added dates to the case list template with a `date_format` used on front end. Added tests for api and search result component.

### CODE REVIEW

File: api/cases/case-list.spec.js [hunk 1]
```diff
@@ -61,22 +63,27 @@ describe('case-list spec', () => {
...
             });
         });
 
         it('should return the columns with one rows', () => {
             return request.get('/api/cases')
                 .expect(200)
                 .then(response => {
-                    expect(response.body.columns.length).toBe(2); // 3 minus case reference
+                    expect(response.body.columns.length).toBe(4); // 5 minus case reference
                     expect(response.body.results.length).toBe(1);
```
[the remark this answers is not in the document]
Comment (mike-suggitt) [reply api/cases/case-list.spec.js]: Yep good spot will remove the column count expects

File: api/cases/sscsCaseList.template.js [hunk 2]
```diff
@@ -4,46 +4,29 @@ module.exports = {
...
             "case_field_id": "parties",
-            "lookup": ["$.appeal.appellant.name.firstName", "$.appeal.appellant.name.lastName", "versus DWP"]
+            "lookup": ["$.case_data.appeal.appellant.name.firstName", "$.case_data.appeal.appellant.name.lastName", "versus DWP"]
         },
         {
             "label": "Type",
             "order": 3,
             "case_field_id": "type",
-            "value": "PIP",
+            "lookup": "PIP",
```
[the remark this answers is not in the document]
Comment (mike-suggitt) [reply api/cases/sscsCaseList.template.js]: Updating to $.case_type_id

[the remark this answers is not in the document]
Comment (mike-suggitt) [reply api/cases/sscsCaseList.template.js]: $.case_type_id return Benefit and not PIP. It doesn't look like BenefitType is currently being returned for cases so probably needs to be hardcoded for now

File: src/app/domain/search-result/search-result.component.spec.ts [hunk 3]
```diff
@@ -11,18 +11,32 @@ import {BrowserTransferStateModule, StateKey} from '@angular/platform-browser';
...
-    'label': 'Type',
-    'order': 3,
-    'case_field_id': 'type',
-    'value': 'PIP',
-
-}];
+        'label': 'Parties',
+        'order': 2,
+        'case_field_id': 'parties',
+        'lookup': ['$.appeal.appellant.name.firstName', '$.appeal.appellant.name.lastName', 'versus DWP']
```
[the remark this answers is not in the document]
Comment (mike-suggitt) [reply src/app/domain/search-result/search-result.component.spec.ts]: I agree it's a bit weird but would suggest we maybe defer this till we can have a bit of a chat. We will need to keep revisiting this as we go on so if happy to leave it as is for now lets get together with @Louisblack and agree what we want to do

[the remark this answers is not in the document]
Comment (cksanders) [reply src/app/domain/search-result/search-result.component.spec.ts]: I have already implemented something along the lines above. Its pretty simple, so possibly we could just amend this now?

[the remark this answers is not in the document]
Comment (mike-suggitt) [reply src/app/domain/search-result/search-result.component.spec.ts]: if it's in your branch then lets merge this and then change as and when your PR goes for review? that way we don't cross-contaminate branches

[the remark this answers is not in the document]
Comment (cksanders) [reply src/app/domain/search-result/search-result.component.spec.ts]: agree to disagree - you lucky its friday!

File: api/cases/sscsCaseList.template.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cksanders) [reply api/cases/sscsCaseList.template.js]: this is odd as the $.case_type_id in the get case response is 'PIP'.

[the remark this answers is not in the document]
Comment (mike-suggitt) [reply api/cases/sscsCaseList.template.js]: there's no concept of "value" in this branch now hence why it was updated to a raw string. there's logic that's been put in that checks for $ and if not it treats it as a value. It's not a massive issue we can look to update using the suffix stuff you've done in your PR if we think it's a better solution going forwards

File: src/app/domain/search-result/search-result.component.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mike-suggitt) [reply src/app/domain/search-result/search-result.component.spec.ts]: :)

File: api/cases/sscsCaseList.template.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cksanders) [reply api/cases/sscsCaseList.template.js]: oh goodness; again lucky its friday. Please update this once my branch is merged.

**Pull request merged**